### DISCUSSION TITLE
增加一个 sing-box-geodata 软件包怎么样

### DISCUSSION THREAD

**Discussion by VergilGao** [Openwrt-Passwall/openwrt-passwall2#495] [Ideas]
我发现一个问题，当singbox开始的时候，他会先下载规则文件，而下载规则文件时他会用默认规则来下载，我比较习惯默认直连，这就导致他下载规则文件失败然后无法正确启动。

但是我手动设置 github.com 之类的网址使用代理，他又匹配不上

```
domain:github.com
domain:githubusercontent.com
```

经过研究，我大概明白了，当没有规则文件时他会下载，有规则文件他就不管了。
那么其实可以仿照v2ray的库那样，也增加一个 sing-box-geodata 软件包，编译时引入，这样就有了默认的规则文件，然后用定时任务去更新。

**Comment by CharlesLocke** [comment]
开发者认为以前那样太臃肿了，所以才改成现在这样。你这是打算又改回去么🤣

**Reply by VergilGao** [reply]
可以不强制依赖，作为可选项。
此外如果要裁剪功能，目前singbox和v2ray-core其实也是完全可以二选一的

**Reply by Deleted-Account** [reply]
确实。现在 xray-core +v2ray-geoip +v2ray-geosite 这三个是强制依赖。有些小空间机子（比如我过年带回老家用的那个小路由，只有128MB的存储空间，虽然平时只做AP），最好还是编译固件的时候，除了核心依赖，app 基本不编译，刷好机子后单独装，强制装装不上的话先装 Xray-core 什么的，装好 Sing-box 后 Xray 那些 再用 --force-depends 单独卸载

**Comment by Travel2Here** [comment]
一般分流用途，这俩文件算是强依赖了吧。我感觉xray那个更新数据功能直接复用给singbox更新就好了。
luci上多个checkbox。体验不是就爽了吗？
